assets/javascripts/discourse/components/discourse-post-event/dates.gjs

```diff
@@ -59,13 +59,15 @@ export default class DiscoursePostEventDates extends Component {
       const result = await cook(this.datesBBCode.join("<span> → </span>"));
       this.htmlDates = htmlSafe(result.toString());
 
-      schedule("afterRender", () => {
-        applyLocalDates(
-          element.querySelectorAll(
-            `[data-post-id="${this.args.event.id}"] .discourse-local-date`
-          ),
-          this.siteSettings
-        );
+      next(() => {
+        schedule("afterRender", () => {
+          applyLocalDates(
+            element.querySelectorAll(
+              `[data-post-id="${this.args.event.id}"] .discourse-local-date`
+            ),
+            this.siteSettings
+          );
+        });
```

Comment:
1. `schedule("afterRender")` inside `next()` is extraneous - `next` already schedules a separate run loop, so specifying that this code needs to run in the "afterRender" part of it shouldn't change anything
2. you might need to add `isDestroying || isDestroyed` check in there so that the async execution doesn't bleed between tests

Reply:
> 1. `schedule("afterRender")` inside `next()` is extraneous - `next` already schedules a separate run loop, so specifying that this code needs to run in the "afterRender" part of it shouldn't change anything

And yet I have 100% repro without, and 0% with this next

Reply:
> 2\. you might need to add `isDestroying || isDestroyed` check in there so that the async execution doesn't bleed between tests

:+1:

Reply:
> 0% with this next

Just to confirm: I'm suggesting using `next` w/o `schedule`, not the other way around. Is `schedule` critical to the fix?

Reply:
> 1. `schedule("afterRender")` inside `next()` is extraneous - `next` already schedules a separate run loop, so specifying that this code needs to run in the "afterRender" part of it shouldn't change anything

Given it fixes my immediate issue I will merge it, but happy to revert if we have a better solution, or if it ends up not fixing it